orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -174,27 +118,23 @@ def backward(ctx, *grad_outputs):
                 for input_name in self._graph_info.user_input_names:
                     # Append to the results the backward output for each input that required grad
                     if input_name in require_grad_names_set:
-                        results.append(_utils._torch_tensor_from_dl_pack(
-                            backward_outputs.dlpack_at(require_grad_names_index),
-                            backward_outputs[require_grad_names_index]))
+                        results.append(backward_outputs[require_grad_names_index])
                         require_grad_names_index += 1
                     else:
                         # input_name is not found in the self._input_info.require_grad_names list
                         # Append None to results for each input that did not require grad
                         results.append(None)
-
+                assert require_grad_names_index == num_user_input_grads
                 # Append gradients of initializer to results
                 # Go over each initializer, check if it required grad and append to results accordingly
                 initializer_index = num_user_input_grads
                 for initializer_name in self._graph_info.initializer_names:
                     if initializer_name in self._graph_initializer_names_to_train:
-                        results.append(_utils._torch_tensor_from_dl_pack(
-                            backward_outputs.dlpack_at(initializer_index),
-                            backward_outputs[initializer_index]))
+                        results.append(backward_outputs[initializer_index])
```

Comment:
what is the reasoning for this change? AFAIK this code was meant to keep high performance